rcl_action/QUALITY_DECLARATION.md

```diff
@@ -0,0 +1,142 @@
+
+### Change Requests [2.i]
+
+This package requires that all changes occur through a pull request.
+
+### Contributor Origin [2.ii]
+
+This package uses DCO as its confirmation of contributor origin policy. More information can be found in [CONTRIBUTING](../CONTRIBUTING.md).
+
+### Peer Review Policy [2.iii]
+
+Following the recommended guidelines for ROS Core packages, all pull requests must have at least 1 peer review.
+
+### Continuous Integration [2.iv]
+
+All pull requests must pass CI on all [tier 1 platforms](https://www.ros.org/reps/rep-2000.html#support-tiers).
+
+### Documentation Policy [2.v]
+
+All pull requests must resolve related documentation changes before merging.
+
+## Documentation [3]
+
+### Feature Documentation [3.i]
+
+`rcl_action` has feature documentation actions in ROS 2.
+It is [hosted](http://design.ros2.org/articles/actions.html).
+
+### Public API Documentation [3.ii]
+
+Most of `rcl_action` has embedded API documentation and it is [hosted](http://docs.ros2.org/latest/api/rcl_action/).
+
+### License [3.iii]
+
+The license for `rcl_action` is Apache 2.0, and a summary is in each source file, the type is declared in the [package.xml](package.xml) manifest file, and a full copy of the license is in the [LICENSE](../LICENSE) file.
+
+There is an automated test which runs a linter that ensures each file has a license statement.
+
+The most recent test results can be found [here](https://ci.ros2.org/view/nightly/job/nightly_linux_release/lastBuild/testReport/rcl_action/copyright/).
+
+### Copyright Statements [3.iv]
+
+The copyright holders each provide a statement of copyright in each source code file in `rcl_action`.
+
+There is an automated test which runs a linter that ensures each file has at least one copyright statement.
+
+The results of the test can be found [here](https://ci.ros2.org/view/nightly/job/nightly_linux_release/lastBuild/testReport/rcl_action/copyright/).
+
+## Testing [4]
+
+### Feature Testing [4.i]
+
+`rcl_action` has feature tests, which test for proper node state transitions.
+The tests are located in the [test](test) subdirectory.
+
+### Public API Testing [4.ii]
+
+Much of the API in `rcl_action` is tested.
+The tests are located in the [test](test) subdirectory.
+
+### Coverage [4.iii]
+
+`rcl_action` does not currently track test coverage.
+
+### Performance [4.iv]
+
+`rcl_action` does not currently have performance tests.
+
+### Linters and Static Analysis [4.v]
+
+`rcl_action` uses and passes all the standard linters and static analysis tools for a C package as described in the [ROS 2 Developer Guide](https://index.ros.org/doc/ros2/Contributing/Developer-Guide/#linters).
+
+Results of the nightly linter tests can be found [here](https://ci.ros2.org/view/nightly/job/nightly_linux_release/lastBuild/testReport/rcl_action).
+
+## Dependencies [5]
+
+### Direct Runtime ROS Dependencies [5.i]/[5.ii]
+
+`rcl_action` has the following runtime ROS dependencies:
+* `action_msgs`
```

Comment:
can we describe the dependencies as it's done [here](https://github.com/ros2/rcl/pull/639/files#diff-5738374578ac1a78c061cd6f5b8dc8f4R143)?

Reply:
Done.